core/objects.go

```diff
@@ -147,36 +147,48 @@ type ValidationRecord struct {
 	AddressesTried []net.IP `json:"addressesTried,omitempty"`
 }
 
-// GetHostPort returns the hostname and port parsed from the inner URL. If the
-// URL doesn't specify a port, the default port for the scheme will be
-// returned. If the URL cannot be parsed, an error will be returned.
+// RehydrateHostPort mutates a validation record. If the URL in the validation
+// record cannot be parsed, an error will be returned. If the Hostname and Port
+// fields already exist in the validation record, they will be retained.
+// Otherwise, the Hostname and Port will be derived from the URL field of the
+// validation record.
 //
 // CABF BRs section 1.6.1: Authorized Ports: One of the following ports: 80
 // (http), 443 (https)
-func (vr ValidationRecord) GetHostPort() (string, string, error) {
+func (vr *ValidationRecord) RehydrateHostPort() error {
+	if vr.URL == "" {
+		return fmt.Errorf("expected URL field in validation record, but could not be found")
+	}
+
+	// Use the existing Hostname and Port from the validation record if it exists.
+	if vr.Hostname != "" && vr.Port != "" {
+		return nil
+	}
```

Comment:
Swap these two. If the hostname and port already exist we want to bail out.

```suggestion
	if vr.URL == "" {
		return fmt.Errorf("expected URL field in validation record, but could not be found")
	}

	// Use the existing Hostname and Port from the validation record if it exists.
	if vr.Hostname != "" && vr.Port != "" {
		return nil
	}
```